### ISSUE TITLE
Launch icon for private browsing

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#419]
### Why/User Benefit/User Problem
As a user, I want to be able to open the browser in private mode for my quick private searches where no history is retained. 

### What / Requirements
Build a solution for a 1-click private mode launch

### Acceptance Criteria (how do I know when I’m done?)
User can launch a search in private mode without opening the regular browser first.

**Comment by brampitoyo** [comment]
I just tested Facebook version 1.9.0 ([Download APK](https://appchive.net/apk/com.facebook.katana/facebook/14776/download)). It was originally built around the time of Android version 4 (Ice Cream Sandwich and Jelly Beans), but can still be installed in Android Pie. It has lots of warnings, but many functionalities were surprisingly intact.

This version of Facebook was special, because it adds three launcher icons for three separate activities: Facebook, Camera and Messenger.

<img width="360" src="https://user-images.githubusercontent.com/51007/52980372-9ee5fc80-343e-11e9-986c-15830468144a.png">

## Pre sign in

Tapping all icons would bring you to the same sign in screen:

<img width="360" src="https://user-images.githubusercontent.com/51007/52980781-baea9d80-3440-11e9-97d3-2169a0d575a5.png">

## Post sign in

* Tapping Facebook would bring you to your timeline on Facebook.app
* Tapping Messenger would bring you to the “Messages” tab on Facebook.app
* Tapping Camera would bring you to a camera view (?) that behaves as if it was a separate app

### Facebook and Messenger

Both launcher icons share the same behaviour:

* They are shortcuts to two views within the same Facebook.app
* If you go to the app switcher, Android System sees one app

Facebook itself promotes these icons not as separate apps, but as “shortcuts for sharing”.

<img width="360" src="https://user-images.githubusercontent.com/51007/52980644-00f33180-3440-11e9-90af-e258ef5dc917.png">

### Camera

This launcher icon behaves differently:

* Distinct, unique UI from Facebook/Messenger
* There’s no entry point to access Facebook/Messenger from Camera, and vice versa
* If I go to the app switcher, Android System thinks that it’s totally different from Facebook/Messenger

In other words, Camera behaves as if it’s a separate app.

![screenshot_20190219-115744](https://user-images.githubusercontent.com/51007/52980611-e4ef9000-343f-11e9-86b5-47599e6a7659.png)

**Comment by brampitoyo** [comment]
### Possible next steps

It seems as if we can design private browsing to behave in two ways:

1. Launcher icon will open private browsing mode within Fenix.app
2. Launcher icon will open private browsing that behaves as if it’s a totally separate app from Fenix

From our recent user research shareout, we know that there’s an opportunity for a specialised private browsing which is good at:

1. Basic browsing performance: simple, lightweight, gets you there quickly
2. Making privacy easy and convenient: no more worries about manually cleaning
3. Creating a clean, distraction-free space: to help you feel organised, productive and tidy

To do this, we may have to pick an approach that defaults to private without requiring forethought or extra steps to switch into a mode within an app.

Is there’s an argument to be made for making private browsing *its own thing*?

**Comment by pocmo** [comment]
Instead of creating a separate launcher icon (and potentially confusing users) could we consider supporting a private browsing **shortcut**? They offer similar functionality AND can be added to the home screen if the user chooses so.

For example with Chrome:

### Shortcuts:

![shortcuts_chrome](https://user-images.githubusercontent.com/89638/53001098-2e34e500-342a-11e9-9200-02cc45615c5b.png)

### Added to home screen:

![shortcuts_chrome](https://user-images.githubusercontent.com/89638/53001107-31c86c00-342a-11e9-9aae-4572168924fb.png)

### Compatibility

* Shortcuts are available since Android 7.1
* Since Android 8.0 shortcuts can be pinned (added to home screen)
* Shortscuts can be dynamic too (e.g. if we want to have shortcuts for sync/linked devices).

**Comment by liuche** [comment]
This is in our Q3, but since it hasn't gone through the UX process yet, maybe we could use it as a case study for UX-Eng early collaboration, @lime124 ? Like @vesta0 decides how long we want to spend on an MVP, and then Eng and UX work together to come up with a solution that fits within that budget.

**Comment by sblatz** [comment]
I'd love to be involved in this 😄

**Comment by liuche** [comment]
We decided to timebox this to 1 sprint.

**Comment by colintheshots** [comment]
I think if we're supporting a full-time Focus Mode, it makes sense to disable the default launcher icon and enable a different one. I've done this before and it works great.

**Comment by andreicristianpetcu** [comment]
Nobody mentioned this but Firefox Light has this feature. 
![IMG_20190720_150838](https://user-images.githubusercontent.com/922368/61578493-5f603100-ab00-11e9-999d-80de562b6dc8.jpg)

**Comment by brampitoyo** [comment]
@andreicristianpetcu I thought that this (otherwise great and useful!) feature has the possibility of introducing some confusion.

When I initially launch Firefox Lite, the app switcher looks like this. So far so good.

<img width="360" src="https://user-images.githubusercontent.com/51007/61598087-9bb0a000-ac6c-11e9-934b-1730a5453f96.png">

Let’s launch Private Browsing Mode. As a user, I notice that the same icon is now used to represent two different apps. Or is it two instances of the same apps?

Even when the icons are identical, each start page has a distinct visual design, so I can still tell the difference between the two modes.

<img width="360" src="https://user-images.githubusercontent.com/51007/61598107-d286b600-ac6c-11e9-8ff3-be5b86912ff1.png">

Finally, let’s open some websites. Can I still tell the difference between the modes? Not easily, but yes.

1. Private mode has a dark toolbar
2. Private mode has a purple mask icon, and different set of icons on its toolbar (e.g. garbage bin)

<img width="360" src="https://user-images.githubusercontent.com/51007/61598299-9dc82e00-ac6f-11e9-846a-bb5507e6f9c1.png">

However, if I activate Night Mode on Firefox Lite, normal browsing now has a dark toolbar too. The only way to tell which mode you’re in is by looking for the purple mask.

<img width="360" src="https://user-images.githubusercontent.com/51007/61598480-a8d08d80-ac72-11e9-8aff-3689c683ca72.png">

-----

Seen from a visual design perspective, solving this problem seems relatively easy:

1. If separated into a different instance, Private Browsing Mode should have a distinct icon.
2. Private Browsing Mode should have a visually distinct appearance when compared to normal browsing – especially on dark theme.

We’ve done (or are in the process of doing) item 2. I’ve designed Private Browsing Mode to have a distinct start page and toolbar.

We haven’t done 1, and I’m wondering whether that’s possible within the constraint of Android OS?

**Comment by lime124** [comment]
removed ux label -it's on our radar already as it's part of the Q3 plan

**Comment by AmyYLee** [comment]
@lime124 sizing UX effort as Medium here.

**Comment by sblatz** [comment]
@AmyYLee to investigate if we need the "toggle" to set "always open in private mode". How does Firefox Lite work? Do we need to add the shortcut to the home screen in order to see the private mode as an option when setting default browser?

**Comment by AmyYLee** [comment]
@sblatz @vesta0 @lime124 I looked into this on Firefox Lite. You don't need a PBM shortcut on homescreen in order to set links to always Open in Fenix PBM as default. In that case, I don't think we need to have the toggle switch in Fenix settings to always open in PBM. I can update the spec to reflect this.

**Comment by vesta0** [comment]
Sounds good @AmyYLee :) Thanks for always thinking about intuitive lean solutions!

**Comment by andreicristianpetcu** [comment]
> "toggle" to set "always open in private mode"

It would be awesome! :D

**Comment by jyeontaek** [comment]
The 3rd party links part is done, but I need more time on the shortcuts after Amy and I had our discussion about pinned shortcuts vs separate launcher.

**Comment by klint** [comment]
Isn't issue #5054 related to this one in a way?
